### PR TITLE
Move secure memory handling to the crypto folder.

### PR DESCRIPTION
**Pull request by ni4**
This PR adds secure_array type, which will be automatically cleared in the destructor, and replaces pgp_forget function with secure_clear. Also it moves all these functions to the crypto folder.

### CODE REVIEW

**Comment by dewyatt** [comment]
I kinda vote against the use of the stack for secure storage, we should be using heap-allocated mlock'd memory ideally. Is the idea to move towards that after this?

**Comment by ni4** [comment]
@dewyatt Yeah, I don't like that too, we should change to use allocated memory. Also we should change this for secure MPIs of the secret keys, which would require some work as `pgp_mpi_t` is now treated as opaque structure. So idea was to first quick-change current code, separating crypto, and later issue PR with more solid updates.

File: src/lib/rnp.cpp [hunk 1]
```diff
@@ -5415,8 +5410,9 @@ try {
     }
 
     /* encrypt secret key if requested */
-    if (op->password) {
-        prov = {.callback = rnp_password_provider_string, .userdata = (void *) op->password};
+    if (op->password.size()) {
```
Comment (dewyatt) [inline src/lib/rnp.cpp:5413]: I think `!op->password.empty()` is more common

File: src/lib/rnp.cpp [hunk 2]
```diff
@@ -5433,11 +5429,7 @@ try {
     }
     ret = RNP_SUCCESS;
 done:
-    if (op->password) {
-        pgp_forget(op->password, strlen(op->password) + 1);
-        free(op->password);
-        op->password = NULL;
-    }
+    op->password.resize(0);
```
Comment (dewyatt) [inline src/lib/rnp.cpp:5432]: Seems odd instead of `op->password.clear()` (should be same effect on modern STLs I believe)

**Review by ronaldtse — APPROVED** [review]
Thanks @ni4! Feel free to merge after addressing the comments!

**Review by antonsviridenko — APPROVED** [review]

File: src/lib/rnp.cpp (hunk 1 above)
Comment (ni4) [reply src/lib/rnp.cpp:5413]: Thanks, @dewyatt. Will fix and force-push it now.

**Comment by ni4** [comment]
Comments addressed, merging.

**Pull request merged**